### PR TITLE
fix: code health cleanup and test gaps (#921, #860)

### LINKED ISSUE #860: [aw][test audit] render_detail: untested branches in `_build_event_details` — SESSION_SHUTDOWN falsy type and TOOL_EXECUTION mod
[Content truncated due to length]
## Summary

Two code branches in `render_detail._build_event_details` are never exercised by the current test suite, so regressions there can't be caught by CI.

---

## Gap 1 — `SESSION_SHUTDOWN` with `shutdownType=None`

**Location:** `src/copilot_usage/render_detail.py:156`

```python
case EventType.SESSION_SHUTDOWN:
    if (data := _safe_event_data(ev, ev.as_session_shutdown)) is None:
        return ""
    return f"type={data.shutdownType}" if data.shutdownType else ""
```

`SessionShutdownData.shutdownType` is typed `str | None`, so the field is legitimately absent in some real events. When the field is `None` or `""`, the function returns `""`.

`TestBuildEventDetails.test_session_shutdown_details` only exercises the truthy case (`shutdownType="routine"`). The **else branch** (`shutdownType is None` → `""`) has zero test coverage.

**Expected test:** An `EVENT_TYPE.SESSION_SHUTDOWN` event with a `SessionShutdownData` whose `shutdownType` is `None` (or omitted from the raw `data` dict) should make `_build_event_details` return `""`.

---

## Gap 2 — `TOOL_EXECUTION_COMPLETE` with non-`None` `model` field

**Location:** `src/copilot_usage/render_detail.py:149–150`

```python
if data.model:
    parts.append(f"model={data.model}")
```

`ToolExecutionData.model` is optional. When present it should appear in the detail string as `model=<name>`. All three existing `TestBuildEventDetails` tests for `TOOL_EXECUTION_COMPLETE` omit the `model` field from the raw dict, so the truthy branch is never reached.

**Expected test:** A `TOOL_EXECUTION_COMPLETE` event whose raw data includes `"model": "claude-sonnet-4"` should produce a details string containing `"model=claude-sonnet-4"`.

---

## Regression risk

Both are **display-path branches** — a silent regression would cause incorrect or missing information in `copilot-usage session <id>` detail output without failing any test. The branches are simple enough that a single parametrized test covers each case.

> Generated by [Test Suite Analysis](https://github.com/microsasa/cli-tools/actions/runs/24139465807/agentic_workflow) · ● 8.6M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Ftest-analysis%22&type=issues)

### LINKED ISSUE #921: [aw][code health] Code hygiene: stale section header, mixed type-ignore styles, and getattr/hasattr in packaging test
## Summary

Three small but real code-hygiene items found during a full codebase review. None are caught by CI (ruff, pyright, bandit). Bundled together because each is a minor nit on its own.

---

## Finding 1 — Dead section header in `models.py`

**Location:** `src/copilot_usage/models.py`, lines 321–323

**Root cause:** A `# ── Typed event helpers ──` block header exists between `GenericEventData` and `# Event envelope` with zero content under it. The typed event helpers that were originally planned for this section were folded into `SessionEvent` as `.as_*()` accessor methods, leaving the section header as a stale artefact.

**Fix:** Remove the three lines (two separator lines + label line, ~lines 321–323).

**Testing:** No regression test needed — the deletion is purely cosmetic and cannot affect runtime behaviour. Confirm `make check` still passes.

---

## Finding 2 — Mixed `# type: ignore` vs `# pyright: ignore` styles

**Location:**
- `src/copilot_usage/cli.py` line 283: `# type: ignore[import-untyped]`
- `src/copilot_usage/cli.py` line 287: `# type: ignore[arg-type]`
- `src/copilot_usage/logging_config.py` line 56: `# type: ignore[arg-type]`

**Correct style in same codebase:**
- `src/copilot_usage/models.py` lines 272, 437: `# pyright: ignore[reportUnknownVariableType]` ✓

**Root cause:** The repo uses pyright in strict mode (no mypy). Suppression comments using mypy error codes (`import-untyped`, `arg-type`) are accepted by pyright but suppress nothing specifically — pyright silently ignores unrecognised mypy codes. This means the suppression is purely cosmetic and doesn't document which pyright diagnostic is being silenced.

**Fix:** Convert to `# pyright: ignore[...]` with the correct pyright diagnostic rule:
- `import-untyped` → `reportMissingModuleSource`
- `arg-type` (cli.py:287, Protocol mismatch) → `reportArgumentType`
- `arg-type` (logging_config.py:56, TypedDict value invariance) → `reportArgumentType`

**Testing:** Run `make check` (typecheck + lint) after the change to confirm no new pyright errors are introduced.

---

## Finding 3 — `getattr` / `hasattr` in `tests/test_packaging.py`

**Location:** `tests/test_packaging.py`, lines 30, 33, 67, 70

**Root cause:** The packaging test uses `getattr(mod, "__all__", None)` and `hasattr(mod, name)` to inspect module public API. The coding guidelines prohibit `getattr`/`hasattr` under the "No Runtime Type Interrogation" rule. The existing `# noqa: B009` comments suppress ruff's B009 rule (getattr with constant attr), but `hasattr` is not covered by B009 at all — those suppressions are inaccurate.

**Fix:** Replace with `vars(mod)` lookups, which return `__dict__` and avoid both `getattr` and `hasattr`:

```python
# Before
all_names = getattr(mod, "__all__", None)  # noqa: B009
...
assert hasattr(mod, name), ...  # noqa: B009

# After
mod_vars = vars(mod)
all_names = mod_vars.get("__all__")
...
assert name in mod_vars, ...
```

**Testing:** Run `make check` (tests pass, ruff clean). The existing packaging test suite itself serves as the regression test — if the rewrite breaks the assertion logic, tests will fail.

---

## Acceptance criteria

- [ ] `models.py` dead section header removed
- [ ] All three `# type: ignore[...]` comments converted to `# pyright: ignore[...]` with correct rule names
- [ ] `tests/test_packaging.py` `getattr`/`hasattr` replaced with `vars(mod)` lookups
- [ ] `make check` passes (lint + typecheck + tests) with no new failures

> Generated by [Code Health Analysis](https://github.com/microsasa/cli-tools/actions/runs/24325053960/agentic_workflow) · ● 7M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Fcode-health%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
## Summary

Addresses **#921** (code hygiene) and **#860** (render_detail test gaps).

### Changes

**#921 — Code hygiene (3 findings):**
- Remove dead `# Typed event helpers` section header in `models.py` (content was moved elsewhere)
- Convert mypy-style `# type: ignore[arg-type]` to `# pyright: ignore[reportArgumentType]` with correct rule names (`cli.py:290`, `logging_config.py:56`)
- Remove unnecessary `# type: ignore[import-untyped]` on watchdog import — pyright doesn't flag it (watchdog ships `py.typed`)
- Replace `getattr`/`hasattr` with `vars()` in `test_packaging.py` for module-level checks, per "no runtime type interrogation" rule
- Intentionally keep `hasattr` for `SessionEvent.parse_data` regression guard — needs full MRO traversal to catch inherited methods on Pydantic models

**#860 — render_detail test gaps:**
- Gap 1 (`shutdownType=None`): Already covered — `shutdownType` is `str` (not `Optional`), defaults to `""`, and Pydantic rejects `None`. Existing tests at lines 2971 and 2984 cover the falsy branch.
- Gap 2 (`TOOL_EXECUTION_COMPLETE` with `model` field): Added `test_tool_execution_with_model_field` — exercises the `if data.model:` branch
- Added `test_tool_execution_with_empty_model_string` — documents falsy `model=""` edge case behaviour

### Review notes
- Triple adversarial review by Codex, Opus 4.6, and Sonnet 4.6
- Opus/Sonnet both caught the `vars(SessionEvent)` MRO weakness — fixed by restoring `hasattr` for that specific assertion
- Sonnet caught the `model=""` edge case gap — test added
- `make check` passes: lint ✅ pyright ✅ bandit ✅ tests 99% ✅ e2e 86 ✅

### Also recommending closure of stale issues
- **#761**: `_detect_resume` was deleted in PR #990; the scenario is covered by `TestOrphanedTurnStartDoesNotInflateActiveCalls`
- **#846**: Already fixed — `test_is_dir_oserror_entry_is_skipped` added in commit 5f456e7

Closes #921
Closes #860

### CODE REVIEW

File: .github/aw/actions-lock.json [hunk 1]
```diff
@@ -14,6 +15,12 @@
       "repo": "github/gh-aw-actions/setup",
       "version": "v0.68.3",
       "sha": "ba90f2186d7ad780ec640f364005fa24e797b360"
+=======
+    "github/gh-aw-actions/setup@v0.60.0": {
+      "repo": "github/gh-aw-actions/setup",
+      "version": "v0.60.0",
+      "sha": "998487a673ace02b3d9586e7511268089af88971"
+>>>>>>> Stashed changes
```
[the remark this answers is not in the document]
Comment (microsasa) [reply .github/aw/actions-lock.json]: Fixed — restored `actions-lock.json` from main. The merge conflict markers were a stale artifact on this branch.

File: tests/test_packaging.py [hunk 2]
```diff
@@ -64,10 +65,11 @@ def test_parse_data_and_event_data_removed_from_public_api() -> None:
...
     assert "EventData" not in dunder_all, "EventData must not be in models.__all__"
-    assert not hasattr(models_mod, "EventData"), (  # noqa: B009
+    models_vars = vars(models_mod)
+    assert "EventData" not in models_vars, (
         "EventData type alias must not exist in models module"
     )
-    assert not hasattr(models_mod.SessionEvent, "parse_data"), (  # noqa: B009
+    assert not hasattr(models_mod.SessionEvent, "parse_data"), (  # noqa: B009  # hasattr intentional: must check full MRO
         "SessionEvent.parse_data() must not exist — use as_*() accessors"
     )
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/test_packaging.py]: Fixed — removed the `# noqa: B009` since B009 only applies to `getattr`, not `hasattr`. Comment now just explains the intentional MRO check.
